cdap-elastic/src/test/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorageTest.java

```diff
@@ -380,4 +380,56 @@ protected void validateCursor(String cursor, int expectedOffset, int expectedPag
     Assert.assertEquals(expectedOffset, c.getOffset());
     Assert.assertEquals(expectedPageSize, c.getLimit());
   }
+
+  @Test
+  public void testSearchReturnsRequiredTerms() throws IOException {
+    MetadataStorage mds = getMetadataStorage();
+    MutationOptions options = MutationOptions.builder().setAsynchronous(false).build();
+
+    List<MetadataRecord> records = IntStream.range(0, 3).boxed().map(i -> new MetadataRecord(
+        MetadataEntity.ofDataset("ns" + i, "ds" + i),
+        new Metadata(MetadataScope.USER, tags("test", "tag", "t" + i),
+            props("p", "v" + i)))).collect(Collectors.toList());
+    mds.batch(records.stream().map(r -> new Update(r.getEntity(), r.getMetadata())).collect(Collectors.toList()),
+        options);
+
+    // return a single required dataset
+    SearchRequest request = SearchRequest.of("+t1").build();
+    SearchResponse response = mds.search(request);
+    Assert.assertEquals(1, response.getResults().size());
+    Assert.assertEquals(new Metadata(MetadataScope.USER, tags("test", "tag", "t1"), props("p", "v1")),
+        response.getResults().get(0).getMetadata());
+
+    // exclude all datasets missing the required term
```

Comment:
Ideally each unit-test method should be small and testing only one case. It's better to break this method into multiple test methods.

Why it is all in one method?

Reply:
I am not sure we have been following that in all of our tests. It is quite tedious and verbose to set up a new test for each case. Many of our unit tests set up the data once, and then run multiple queries against that data.

Reply:
It depends on how expensive the setup is. Not all existing unit tests are good examples, especially those older one. Breaking it down into small test methods make it easier to reason what is being tested.

Reply:
Originally, my test cases were set up in such a way as to resemble the previous test cases in the file as closely as possible.
On my machine, each of the test cases introduced--when split up as suggested--takes around 2 seconds to run.